presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/thrift/ThriftMetastoreUtil.java

```diff
@@ -478,13 +480,30 @@ private static SerDeInfo getSerdeInfo(org.apache.hadoop.hive.metastore.api.Table
         return serdeInfo;
     }
 
+    // extract lastDataCommitTime from partition params.
+    public static int getLastDataCommitTimeFromParams(org.apache.hadoop.hive.metastore.api.Partition partition)
+    {
+        if (!partition.isSetParameters()) {
+            return 0;
+        }
+
+        Map<String, String> params = partition.getParameters();
+        try {
+            return Integer.parseInt(params.getOrDefault(LAST_DATA_COMMIT_TIME, "0"));
+        }
+        catch (NumberFormatException e) {
+            return 0;
+        }
+    }
+
     public static Partition fromMetastoreApiPartition(org.apache.hadoop.hive.metastore.api.Partition partition, PartitionMutator partitionMutator, ColumnConverter columnConverter)
     {
         StorageDescriptor storageDescriptor = partition.getSd();
         if (storageDescriptor == null) {
             throw new PrestoException(HIVE_INVALID_METADATA, "Partition does not contain a storage descriptor: " + partition);
         }
 
+        int lastDataCommitTime = getLastDataCommitTimeFromParams(partition);
```

Comment:
comment not addressed: https://github.com/prestodb/presto/pull/17830/files#r898716046